ddht/v5_1/network.py

```diff
@@ -540,6 +541,53 @@ async def _monitor_done(
     )
 
 
+@asynccontextmanager
+async def common_network_stream_find_nodes(
+    network: NetworkAPI,
+    node_id: NodeID,
+    endpoint: Endpoint,
+    distances: Collection[int],
+    *,
+    request_id: Optional[bytes] = None,
+) -> AsyncIterator[trio.abc.ReceiveChannel[ENRAPI]]:
+    async def _stream_find_nodes_response(
+        send_channel: trio.abc.SendChannel[ENRAPI],
+    ) -> None:
+        async with network.client.stream_find_nodes(
+            node_id, endpoint, distances=distances, request_id=request_id
+        ) as resp_aiter:
+            async with send_channel:
+                async for response in resp_aiter:
+                    enrs = response.message.enrs
+                    for enr in enrs:
+                        try:
+                            await send_channel.send(enr)
+                        except (trio.BrokenResourceError, trio.ClosedResourceError):
+                            break
+
+    if not distances:
```

Comment:
*nitpick* maybe move this above the function definition since we typically want to error out as early as possible.